centrifuge-app/src/components/Charts/PoolPerformanceChart.tsx

```diff
@@ -255,35 +382,100 @@ function PoolPerformanceChart() {
 
 function CustomLegend({
   data,
+  setRange,
+  selectedTabIndex,
 }: {
   data: {
+    currency: string
     nav: number
-    price: number | null
+    juniorTokenPrice: number
+    seniorTokenPrice?: number | null
+    juniorAPY: number
+    seniorAPY: number
   }
+  setRange: (value: { value: string; label: string }) => void
+  selectedTabIndex: number
 }) {
-  const theme = useTheme()
+  const Dot = ({ color }: { color: string }) => (
+    <Box width="8px" height="8px" borderRadius="50%" backgroundColor={color} marginRight="4px" />
+  )
+
+  const navObj = {
```

Comment:
```suggestion
  const navData = {
```
for consistency though out